disnake/abc.py

```diff
@@ -1348,27 +1348,38 @@ async def send(
         state = self._state
         content = str(content) if content is not None else None
 
+        if file is not None and files is not None:
+            raise InvalidArgument("cannot pass both file and files parameter to send()")
+
+        if file is not None:
+            if not isinstance(file, File):
+                raise InvalidArgument("file parameter must be File")
+            files = [file]
+
         if embed is not None and embeds is not None:
             raise InvalidArgument("cannot pass both embed and embeds parameter to send()")
 
         if embed is not None:
-            embed = embed.to_dict()
+            embeds = [embed]
 
-        elif embeds is not None:
+        if embeds is not None:
             if len(embeds) > 10:
                 raise InvalidArgument("embeds parameter must be a list of up to 10 elements")
+            for embed in embeds:
+                if embed._files:
+                    files = files or []
+                    files += embed._files
```

Comment:
Everything in this PR is ok except this code pattern appearing several times in different files, but it's mostly the problem of send/edit methods implementation